tasks/prepare-fbc-release/prepare-fbc-release.yaml

```diff
@@ -0,0 +1,114 @@
+            echo "No data JSON was provided."
+            exit 1
+        fi        
+
+        pattern="^v[0-9]+\.[0-9]+$"
+
+        # Read components and initial values
+        components=$(jq -c '.components[]' "$SNAPSHOT_PATH")
+        fromIndex=$(jq -r '.fbc.fromIndex' "$DATA_FILE")
+        targetIndex=$(jq -r '.fbc.targetIndex' "$DATA_FILE")
+        binaryImage=$(jq -r '.fbc.binaryImage' "$DATA_FILE")
+
+        # Print initial values
+        echo "Initial fromIndex: $fromIndex"
+        echo "Initial targetIndex: $targetIndex"
+        echo "Initial binaryImage: $binaryImage"
+        echo
+
+        # Count components
+        COMPONENT_COUNT=$(jq '.components | length' "$SNAPSHOT_PATH")
+        echo "found $COMPONENT_COUNT components"
+
+        # Initialize jq_script
+        jq_script="."
+
+        # Function to replace tag in an image
+        replace_tag() {
+            local updatedImage="${1%:*}:${2}"
+            echo "$updatedImage"
+        }
+
+        # Iterate over components
+        while read -r component; do
+            containerImage=$(echo "$component" | jq -r '.containerImage')
+            componentName=$(echo "$component" | jq -r '.name')
+
+            # Fetch OCP version using skopeo
+            ocpVersion=$(skopeo inspect --raw docker://"$containerImage" \
+                | jq -r '.annotations."org.opencontainers.image.base.name"' | cut -d: -f2)
+
+            # Check if the version matches the pattern
+            if ! echo "$ocpVersion" | grep -Eq "$pattern"; then
+                echo "Invalid format for image $containerImage."
+                continue
+            fi
+
+            # Compute updated values
+            updatedFromIndex=$(replace_tag "$fromIndex" "$ocpVersion")
+            updatedTargetIndex=$(replace_tag "$targetIndex" "$ocpVersion")
+            updatedBinaryImage=$(replace_tag "$binaryImage" "$ocpVersion")
+
+            # Print updated values
+            echo "Component: $componentName"
+            echo "ocpVersion: $ocpVersion"
+            echo "Updated fromIndex for $componentName: $updatedFromIndex"
+            echo "Updated targetIndex for $componentName: $updatedTargetIndex"
+            echo "Updated binaryImage for $componentName: $updatedBinaryImage"
+            echo
+
+            # Build jq script to update JSON
+            jq_script+=" | (.components[] | select(.name == \"$componentName\") | .ocpVersion) |= \"$ocpVersion\""
+            jq_script+=" | (.components[] | select(.name == \"$componentName\") | .updatedFromIndex) |= \"$updatedFromIndex\""
+            jq_script+=" | (.components[] | select(.name == \"$componentName\") | .updatedTargetIndex) |= \"$updatedTargetIndex\""
+            jq_script+=" | (.components[] | select(.name == \"$componentName\") | .updatedBinaryImage) |= \"$updatedBinaryImage\""
+        done < <(echo "$components")
+
+        # Apply updates to JSON
+        updated_json=$(jq "$jq_script" "$SNAPSHOT_PATH")
```

Comment:
In theory this should not work as the `jq_script` is only appended inside the subshell.  https://mywiki.wooledge.org/BashFAQ/024

Reply:
Which subshell? There is no subshell. The article you linked to talks about using a pipe. For example:

`true | { myvar=bla;  }; echo $myvar` - this won't work

`{ myvar=bla; }; echo $myvar` - this will work

Reply:
You are completely right. I did not notice the `|` there. Thanks for the correction.